frontend/src/pages/offers/ViewOfferPage.tsx

```diff
@@ -1,27 +1,70 @@
-import { FunctionComponent } from 'react'
+import { FunctionComponent, useState } from 'react'
 import { useParams } from 'react-router-dom'
+import Badge from '../../components/Badge'
+import Button from '../../components/Button'
 import ReadOnlyField from '../../components/forms/ReadOnlyField'
 import InternalLink from '../../components/InternalLink'
 import LayoutWithNav from '../../layouts/LayoutWithNav'
-import { useOfferQuery, useShipmentQuery } from '../../types/api-types'
+import {
+  OfferDocument,
+  PalletCreateInput,
+  PalletType,
+  useCreatePalletMutation,
+  useGroupQuery,
+  useOfferQuery,
+  useShipmentQuery,
+} from '../../types/api-types'
 import { formatShipmentName } from '../../utils/format'
 import { shipmentViewOffersRoute } from '../../utils/routes'
+import CreatePalletModal from './CreatePalletModal'
 
 const ViewOfferPage: FunctionComponent = () => {
   const params = useParams<{ shipmentId: string; offerId: string }>()
   const offerId = parseInt(params.offerId, 10)
   const shipmentId = parseInt(params.shipmentId, 10)
 
-  // Figure out the shipment ID from the path
-  const { data: targetShipment } = useShipmentQuery({
+  const { data: shipment } = useShipmentQuery({
     variables: { id: shipmentId },
   })
 
-  const { data } = useOfferQuery({ variables: { id: offerId } })
+  const { data: offer } = useOfferQuery({ variables: { id: offerId } })
+
+  const sendingGroupId = offer?.offer.sendingGroupId
+  const { data: sendingGroup } = useGroupQuery({
+    skip: !sendingGroupId, // Don't execute this query until the group ID is known
```

Comment:
This is pretty cool:
1. we load some details about the offer on line 30
2. this gives us the sendingGroupId
3. once we have that ID, we fetch some details about the group

We could probably change the backend so that it returns some details about the sending group when querying the offer, but this works well anyway.

Reply:
Oh nice, yeah this is pretty convenient.

We could make the offer query support returning the offer's group pretty easily too.